elliottlib/cli/find_builds_cli.py

```diff
@@ -182,9 +183,12 @@ def _fetch_builds_by_kind_image(runtime, default_product_version, session):
         image_metadata)
 
     # Returns a list of (n, v, r, pv) tuples of each build
+    latest_builds = []
+    tag_set = set()
+    brew_session = koji.ClientSession(constants.BREW_HUB)
     image_tuples = parallel_results_with_progress(
         image_metadata,
-        lambda build: build.get_latest_build_info(product_version_overide)
+        lambda build: build.get_latest_build_info(product_version_overide, tag_set, latest_builds, brew_session)
```

Comment:
- I didn't introduce this `parallel_results_with_progress` and a lot of our functions are using it, I could help to take a look if they are safe or not.
- because for 4.2+ there's are two tags, it's not over and over again, it's only two calls, but if we have 100+ tags, then it's 100+ calls because AFAIK I didn't find a API call pass in multiple tags and return a single list of builds, let me know if you find any.

Reply:
The `parallel_results_with_progress` function will run callback function in parallel. It doesn't have race condition before because it writes nothing to a shared variable, but this PR will write values to both `tag_set` and `latest_builds`. A multithreaded read-modify-write sequence to a non-atomic data structure without lock is usually a race condition.